transport/src/test/java/io/netty/channel/pool/SimpleChannelPoolTest.java

```diff
@@ -242,4 +244,124 @@ public void initChannel(LocalChannel ch) throws Exception {
+                    ch.pipeline().addLast(new ChannelInboundHandlerAdapter());
+                }
+            });
+
+        // Start server
+        Channel sc = sb.bind(address).sync().channel();
+        CountingChannelPoolHandler handler = new CountingChannelPoolHandler();
+
+        int count = 256;
+
+        SimpleChannelPool pool = new SimpleChannelPool(cb, handler);
+
+        ConcurrentSet<Channel> concurrentSet = new ConcurrentSet<Channel>();
+
+        for (int i = 0; i < count; ++ i) {
+            Future<Channel> future = pool.acquire();
+
+            try {
+                future.get();
+            } catch (Exception e) {
+                Assert.assertTrue(false);
+            }
+
+            concurrentSet.add(future.get());
+        }
+
+        Assert.assertTrue(concurrentSet.size() == count);
+
+        pool.close();
+
+        Assert.assertTrue(pool.allChannels.size() == 0);
+
+        for (Channel channel : concurrentSet) {
+            Assert.assertTrue(! channel.isActive());
+        }
+
+        sc.close().sync();
+        group.shutdownGracefully();
+    }
+
+    @Test
+    public void testClose2() throws Exception {
+        EventLoopGroup group = new LocalEventLoopGroup();
+        LocalAddress address = new LocalAddress(LOCAL_ADDR_ID + ".SimpleChannelPoolTest.testClose2");
+        Bootstrap cb = new Bootstrap();
+        cb.remoteAddress(address);
+        cb.group(group).channel(LocalChannel.class);
+
+        ServerBootstrap sb = new ServerBootstrap();
+        sb.group(group).channel(LocalServerChannel.class).childHandler(
+            new ChannelInitializer<LocalChannel>() {
+                @Override
+                public void initChannel(LocalChannel ch) throws Exception {
+                    ch.pipeline().addLast(new ChannelInboundHandlerAdapter());
+                }
+            });
+
+        // Start server
+        Channel sc = sb.bind(address).sync().channel();
+        CountingChannelPoolHandler handler = new CountingChannelPoolHandler();
+
+        int count = 256;
+
+        SimpleChannelPool pool = new SimpleChannelPool(cb, handler);
+
+        ConcurrentSet<Channel> concurrentSet = new ConcurrentSet<Channel>();
+
+        for (int i = 0; i < count; ++ i) {
+            Future<Channel> future = pool.acquire();
+
+            try {
+                future.get();
+            } catch (Exception e) {
+                Assert.assertTrue(false);
+            }
+
+            concurrentSet.add(future.get());
+        }
+
+        Assert.assertTrue(concurrentSet.size() == count);
+
+        for (Channel channel : concurrentSet) {
+            try {
+                channel.close().sync();
+            } catch (Exception e) {
+                Assert.assertTrue(false);
+            }
+
+            Assert.assertTrue(! channel.isActive());
+
+            try {
+                pool.release(channel).sync();
+            } catch (Exception e) {
```

Comment:
I use syn to make sure that channel is released. And in this case channel is not health and release will fail.